storm-core/src/jvm/org/apache/storm/container/cgroup/core/BlkioCore.java

```diff
@@ -0,0 +1,259 @@
+        }
+        return result;
+    }
+
+    public void setReadBps(Device device, long bps) throws IOException {
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_READ_BPS_DEVICE), makeContext(device, bps));
+    }
+
+    public Map<Device, Long> getReadBps() throws IOException {
+        List<String> strings = CgroupUtils.readFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_READ_BPS_DEVICE));
+        Map<Device, Long> result = new HashMap<Device, Long>();
+        for (String string : strings) {
+            String[] strArgs = string.split(" ");
+            Device device = new Device(strArgs[0]);
+            Long bps = Long.valueOf(strArgs[1]);
+            result.put(device, bps);
+        }
+        return result;
+    }
+
+    public void setWriteBps(Device device, long bps) throws IOException {
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_WRITE_BPS_DEVICE), makeContext(device, bps));
+    }
+
+    public Map<Device, Long> getWriteBps() throws IOException {
+        List<String> strings = CgroupUtils.readFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_WRITE_BPS_DEVICE));
+        Map<Device, Long> result = new HashMap<Device, Long>();
+        for (String string : strings) {
+            String[] strArgs = string.split(" ");
+            Device device = new Device(strArgs[0]);
+            Long bps = Long.valueOf(strArgs[1]);
+            result.put(device, bps);
+        }
+        return result;
+    }
+
+    public void setReadIOps(Device device, long iops) throws IOException {
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_READ_IOPS_DEVICE), makeContext(device, iops));
+    }
+
+    public Map<Device, Long> getReadIOps() throws IOException {
+        List<String> strings = CgroupUtils.readFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_READ_IOPS_DEVICE));
+        Map<Device, Long> result = new HashMap<Device, Long>();
+        for (String string : strings) {
+            String[] strArgs = string.split(" ");
+            Device device = new Device(strArgs[0]);
+            Long iops = Long.valueOf(strArgs[1]);
+            result.put(device, iops);
+        }
+        return result;
+    }
+
+    public void setWriteIOps(Device device, long iops) throws IOException {
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_WRITE_IOPS_DEVICE), makeContext(device, iops));
+    }
+
+    public Map<Device, Long> getWriteIOps() throws IOException {
+        List<String> strings = CgroupUtils.readFileByLine(Constants.getDir(this.dir, BLKIO_THROTTLE_WRITE_IOPS_DEVICE));
+        Map<Device, Long> result = new HashMap<Device, Long>();
+        for (String string : strings) {
+            String[] strArgs = string.split(" ");
+            Device device = new Device(strArgs[0]);
+            Long iops = Long.valueOf(strArgs[1]);
+            result.put(device, iops);
+        }
+        return result;
+    }
```

Comment:
`getBlkioWeightDevice`, `getReadBps`, `getWriteBps`, `getReadIOps`, `getWriteIOps`

Don't Repeat Yourself. Same with the set versions.